apps/explorer/src/app/components/txs/details/tx-details-wrapper.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import { DATA_SOURCES } from '../../../config';
+import { t, useFetch } from '@vegaprotocol/react-helpers';
+import { TxDetailsOrder } from './tx-order';
+import type { BlockExplorerTransactionResult } from '../../../routes/types/block-explorer-response';
+import type { TendermintBlocksResponse } from '../../../routes/blocks/tendermint-blocks-response';
+import { TxDetailsHeartbeat } from './tx-hearbeat';
+import { TxDetailsLPAmend } from './tx-lp-amend';
+import { TxDetailsGeneric } from './tx-generic';
+import { TxDetailsBatch } from './tx-batch';
+import { TxDetailsChainEvent } from './tx-chain-event';
+import { TxContent } from '../../../routes/txs/id/tx-content';
+
+interface TxDetailsWrapperProps {
+  txData: BlockExplorerTransactionResult | undefined;
+  pubKey: string | undefined;
+  height: string;
+}
+
+export const TxDetailsWrapper = ({
+  txData,
+  pubKey,
+  height,
+}: TxDetailsWrapperProps) => {
+  const {
+    state: { data: blockData },
+  } = useFetch<TendermintBlocksResponse>(
+    `${DATA_SOURCES.tendermintUrl}/block?height=${height}`
+  );
+
+  if (!txData) {
+    return <>{t('Awaiting Block Explorer transaction details')}</>;
+  }
+
+  let child;
+
+  if (txData.type === 'Submit Order') {
```

Comment:
nitpicky suggestion (non-blocking): i think a switch statement would look more neat here, intead of the if/else chain, but this is also subjective.

Reply:
I agree and have done this